src/sentry/digests/backends/redis.py

```diff
@@ -448,132 +198,51 @@ def maintenance(self, deadline, chunk=1000):
+            try:
+                response = script(
+                    connection, [key], ['DIGEST_OPEN', self.namespace, self.ttl, timestamp, key]
+                )
+            except ResponseError as e:
+                if 'err(invalid_state):' in e.message:
+                    six.raise_from(
+                        InvalidState('Timeline is not in the ready state.'),
+                        e,
                     )
-                    pipeline.execute()
-
-            def unschedule():
-                with connection.pipeline() as pipeline:
-                    # Watch the timeline to ensure that no other transactions add
-                    # events to the timeline while we are trying to delete it.
-                    pipeline.watch(timeline_key)
-                    pipeline.multi()
-                    if connection.zcard(timeline_key) == 0:
-                        cleanup_records(pipeline)
-                        pipeline.delete(make_last_processed_timestamp_key(timeline_key))
-                        pipeline.zrem(make_schedule_key(self.namespace, SCHEDULE_STATE_READY), key)
-                        pipeline.zrem(
-                            make_schedule_key(self.namespace, SCHEDULE_STATE_WAITING), key
-                        )
-                        pipeline.execute()
-
-            # If there were records in the digest, we need to schedule it so
-            # that we schedule any records that were added during digestion. If
-            # there were no items, we can try to remove the timeline from the
-            # digestion schedule.
-            if records:
-                reschedule()
-            else:
-                try:
-                    unschedule()
-                except WatchError:
-                    logger.debug('Could not remove timeline from schedule, rescheduling instead')
-                    reschedule()
-
-    def delete(self, key):
-        timeline_key = make_timeline_key(self.namespace, key)
-
-        connection = self.cluster.get_local_client_for_key(timeline_key)
-
-        lock = self.locks.get(timeline_key, duration=30, routing_key=timeline_key)
-        with lock.acquire(), connection.pipeline() as pipeline:
-            clear_timeline_contents(pipeline, timeline_key)
-            pipeline.zrem(make_schedule_key(self.namespace, SCHEDULE_STATE_READY), key)
-            pipeline.zrem(make_schedule_key(self.namespace, SCHEDULE_STATE_WAITING), key)
-            pipeline.execute()
+                else:
+                    raise
+
+            yield filter(
+                lambda record: record.value is not None,
+                map(
+                    lambda item: Record(
+                        item[0],
+                        self.codec.decode(item[1]) if item[1] is not None else None,
+                        float(item[2]),
+                    ),
+                    response,
+                )
+            )
```

Comment:
I think I just got lazy here, I'll make this bit clearer (this should also address the above comment.)